x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/MlInitializationService.java

```diff
@@ -112,5 +142,113 @@ MlDailyMaintenanceService getDailyMaintenanceService() {
         return mlDailyMaintenanceService;
     }
 
+    private void makeMlIndicesHidden() {
+        String[] mlHiddenIndexPatterns = MachineLearning.getMlHiddenIndexPatterns();
+
+        CountDownLatch latch = new CountDownLatch(1);
+
+        LatchedActionListener<Boolean> listener = new LatchedActionListener<>(
+            ActionListener.wrap(
+                result -> logger.info("All the ML internal indices and aliases were successfully made hidden"),
```

Comment:
I think this will get a bit tiresome, as people will see it in the logs every time a new master node is elected for the next 3 years.  I think the response should be changed so that `true` means log this message and `false` means don't log this message.  Then the response should be set to `true` if something was changed successfully and `false` if either nothing was changed _or_ a more detailed error message has already been logged.

Then, in the case where this method doesn't make any changes, which will be by far the most common thing to happen when a master node is elected over the next 3 years, this method should log nothing at `info` level or higher.